easy-ui-react/src/TabPanels/TabPanelsTabs.tsx

```diff
@@ -0,0 +1,79 @@
+import { AriaLabelingProps, ItemElement, Node } from "@react-types/shared";
+import React, { useEffect } from "react";
+import { mergeProps, useTab, useTabList } from "react-aria";
+import { TabListState, useTabListState } from "react-stately";
+import { Tabs } from "../Tabs";
+import { useTabPanels } from "./context";
+
+type TabPanelsTabsProps = AriaLabelingProps & {
+  /**
+   * The tab items to display. Item keys should match the key of the
+   * corresponding `<Item>` within the `<TabPanels.Panels>` element.
+   */
+  children: ItemElement<object> | ItemElement<object>[];
+};
+
+type TabPanelsTabProps = {
+  /** The item for the tab. */
+  item: Node<object>;
+
+  /** The state for the tab. */
+  state: TabListState<object>;
+};
+
+/**
+ * Represents a container of tabs in a `<TabPanels />`.
+ */
+export function TabPanelsTabs(props: TabPanelsTabsProps) {
+  const { tabProps, setTabListState } = useTabPanels();
+  const ref = React.useRef(null);
+  const mergedProps = mergeProps(tabProps, props);
+  const state = useTabListState(mergedProps);
+  const { tabListProps } = useTabList(mergedProps, state, ref);
+
+  useEffect(
+    () => {
+      setTabListState(state);
+    },
+    // state is too comprehensive of a dependency in and of itself. only change
```

Comment:
**question**: by comprehensive, do you mean there are too many dependencies of state to include here or that the other ones aren't really useful for us?

Reply:
yeah listening to the entire state causes a render loop because it is concerned with fields that don't apply to the tab list on this component. this just listens for the ones we care about. i updated the comment to reflect that so it's more intuitive